tests/drizzle/test_kernel.py

```diff
@@ -0,0 +1,140 @@
+import pytest
+import numpy as np
+from astropy import wcs
+from drizzlepac import cdriz
+import cdriz_setup
+
+
+@pytest.fixture
+def kernel_pars():
+    _params = cdriz_setup.Get_Grid(inx=50, iny=60, outx=51, outy=66)
+    _params.zero_background()
+    return _params
+
+
+@pytest.mark.parametrize("kernel", ["square", "point", "turbo", "gaussian", "lanczos3"])
+def test_point_kernel(kernel, kernel_pars, new_truth=False, return_png=False):
+    """Function tests different c code point kernels (inputs already created on instantiation).
+
+    Parameters
+    ----------
+    kernel : str
+        String associated with one of the c code point kernel options.
+    kernel_pars : Class
+        The Class inintialized in Get_Grid which includes all of the inputs need to run cdriz.tdriz.
+    new_truth : bool, optional
+        Flag, whether to save a new csv truth file from the output data, by default False
+    return_png: bool, optional
+        Flag, whether to create an output image map (png).
+    """
+    # truth filename
+    truth_filename = f"./tests/drizzle/truth_files/{kernel}_truth"
+
+    # add missing/flagged pixels in inwht
+    kernel_pars.insci[20:22, 21:22] = 100
+
+    # resample:
+    cdriz_setup.cdriz_call(kernel_pars, kernel)
+
+    if new_truth:
+        # save new truth file
+        cdriz_setup.save_array(kernel_pars.outsci, f"{truth_filename}.csv")
+
+    if return_png:
+        # save truth file as figure
+        cdriz_setup.generate_png(kernel_pars, f"{truth_filename}.png")
+
+    truth_array = np.genfromtxt(f"{truth_filename}.csv", delimiter=",")
+
+    wrong_mask = np.logical_not(np.isclose(kernel_pars.outsci, truth_array))
+
+    assert np.allclose(
+        kernel_pars.outsci,
+        truth_array,
+        atol=1e-4,
+        rtol=1e-5,
+    ), cdriz_setup.error_message(kernel_pars.outsci, f"{truth_filename}_new.csv")
+    # np.testing.assert_array_equal(kernel_pars.outsci, truth_array)
+
+
+def test_cdriz_edge(kernel_pars, kernel="gaussian", new_truth=False, return_png=False):
+    """Similar to test_point_kernel but looking at bright pixels at edge of field."""
+
+    truth_filename = f"./tests/drizzle/truth_files/edge_{kernel}_truth"
+    kernel_pars.insci[0, 21] = 100
+    cdriz_setup.cdriz_call(kernel_pars, kernel)
+    if new_truth:
+        cdriz_setup.save_array(kernel_pars.outsci, f"{truth_filename}.csv")
+    if return_png:
+        cdriz_setup.generate_png(kernel_pars, f"{truth_filename}.png")
+
+    truth_array = np.genfromtxt(f"{truth_filename}.csv", delimiter=",")
+    assert np.allclose(
+        kernel_pars.outsci, truth_array, atol=1e-4
+    ), cdriz_setup.error_message(kernel_pars.outsci, f"{truth_filename}_new.csv")
+
+
+def test_cdriz_large(kernel_pars, kernel="gaussian", new_truth=False, return_png=False):
+    """Similar to test_point_kernel but looking at large pixel."""
+
+    truth_filename = f"./tests/drizzle/truth_files/large_sqaure_{kernel}_truth"
+    kernel_pars.insci[21:25, 22:26] = 100
```

Comment:
Again, directory path construction to avoid hard-coding.  Also, I think you mean "square" and not "sqaure".